contracts/aux/auto-alex-v3-wrapped.clar

```diff
@@ -0,0 +1,107 @@
+(define-data-var token-symbol (string-ascii 32) "vLiALEX")
+(define-data-var token-uri (optional (string-utf8 256)) (some u"https://cdn.alexlab.co/metadata/auto-alex-v3-wrapped.json"))
+
+;; governance functions
+
+(define-public (set-name (new-name (string-ascii 32)))
+	(begin
+		(try! (is-dao-or-extension))
+		(ok (var-set token-name new-name))))
+
+(define-public (set-symbol (new-symbol (string-ascii 32)))
+	(begin
+		(try! (is-dao-or-extension))
+		(ok (var-set token-symbol new-symbol))))
+
+(define-public (set-token-uri (new-uri (optional (string-utf8 256))))
+	(begin
+		(try! (is-dao-or-extension))
+		(ok (var-set token-uri new-uri))))
+
+;; public functions
+
+(define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
+	(begin
+		(asserts! (or (is-eq tx-sender sender) (is-eq contract-caller sender)) err-unauthorised)
+		(try! (ft-transfer? auto-alex-v3-wrapped amount sender recipient))
+		(print { type: "transfer", amount: amount, sender: sender, recipient: recipient, memo: memo })
+		(ok true)))
+
+(define-public (transfer-fixed (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
+    (transfer amount sender recipient memo))
+
+(define-public (mint (amount uint) (recipient principal))
+	(begin 
+		(asserts! (or (is-eq tx-sender recipient) (is-eq contract-caller recipient)) err-unauthorised)				
+		(try! (ft-mint? auto-alex-v3-wrapped (get-tokens-to-shares amount) recipient))
+		(contract-call? .auto-alex-v3 transfer amount recipient (as-contract tx-sender) none)))
+
+(define-public (mint-fixed (amount uint) (recipient principal))
+    (mint amount recipient))
+
+(define-public (burn (amount uint) (sender principal))
+	(begin
+		(asserts! (or (is-eq tx-sender sender) (is-eq contract-caller sender)) err-unauthorised)
+		(as-contract (try! (contract-call? .auto-alex-v3 transfer (get-shares-to-tokens amount) tx-sender sender none)))
+		(ft-burn? auto-alex-v3-wrapped amount sender)))
+
+(define-public (burn-fixed (amount uint) (sender principal))
+    (burn amount sender))
+
+;; read-only functions
+
+(define-read-only (is-dao-or-extension)
+	(ok (asserts! (or (is-eq tx-sender .lisa-dao) (contract-call? .lisa-dao is-extension contract-caller)) err-unauthorised)))
+	
+(define-read-only (get-name)
+	(ok (var-get token-name)))
+
+(define-read-only (get-symbol)
+	(ok (var-get token-symbol)))
+
+(define-read-only (get-token-uri)
+	(ok (var-get token-uri)))
+
+(define-read-only (get-decimals)
+	(ok token-decimals))
+
+(define-read-only (get-balance (who principal))
+	(ok (ft-get-balance auto-alex-v3-wrapped who)))
+
+(define-read-only (get-balance-fixed (who principal))
+	(get-balance who))
+
+(define-read-only (get-total-supply)
+	(ok (ft-get-supply auto-alex-v3-wrapped)))
+
+(define-read-only (get-total-supply-fixed)
+	(get-total-supply))
+
+(define-read-only (get-share (who principal))
+	(ok (get-shares-to-tokens (unwrap-panic (get-balance who)))))
```

Comment:
use `(ft-get-balance auto-alex-v3-wrapped who)` directly

Reply:
https://github.com/lisalab-io/liquid-stacking/pull/56/commits/20107362b60f7890c0538305e7089746f26b6ac3

Reply:
I don't think so, because that gives you `get-balance`, while `get-share` gives you the underlying rebase tokens held by the wrapped token.

Reply:
(get-balance) is equal to (ft-get-balance auto-alex-v3-wrapped who)

Reply:
https://github.com/lisalab-io/liquid-stacking/pull/56/commits/389890350ed3123517d8aeaebd3222b1669c3661